blog/_posts/2018-03-29-missing.md

```diff
@@ -0,0 +1,431 @@
+described above. The only difference is that it uses a `BitArray` rather than an
+`Array{UInt8}` to indicate whether a value is missing, therefore taking 1 bit per
+entry rather than 8 bits. Even if it consumes more memory, the `Array{UInt8}` mask
+approach is faster (at least in the current state of `BitArray`), and it generalizes
+to `Union`s of more than two types. However, we are aware that other implementations
+such as [PostgreSQL](https://www.postgresql.org/docs/9.5/static/storage-page-layout.html)
+or [Apache Arrow](https://arrow.apache.org/docs/memory_layout.html#null-bitmaps)
+use bitmaps equivalent to `BitArray`.
+
+## Current limitations and future developments
+
+As illustrated above, the new `missing` representation satisfies most requirements for a
+technical and scientific language like Julia. However, it suffers from some limitations,
+some temporary and some more fundamental.
+
+One area which could be improved easily concerns convenience syntax and functions
+to deal with missing values. We are fully aware that `Union{Missing,T}` is quite verbose
+for those using missing values in daily work. The `T?` syntax has been discussed
+as a compact alternative, inspired by languages with `Nullable` types. However it is
+not clear yet whether it would be more appropriate to attribute this syntax to
+`Union{Missing,T}` or to `Union{Nothing,T}`. It is therefore currently reserved waiting
+for a decision. A possible solution would be to introduce one dedicated syntax for each
+of these types.
+
+Convenience functions would also be useful to propagate missing values with functions
+which have not been written to do it automatically. Constructs like `lift(f, x)`,
+`lift(f)(x)` and `f?(x)` have been [discussed](https://github.com/JuliaLang/julia/pull/26661)
+to provide a shorter equivalent of `ismissing(x) ? missing : f(x)`.
+
+Other temporary limitations concern compiler optimizations which are not yet implemented.
+First, function calls involving missing values [are currently](https://github.com/JuliaLang/julia/issues/23338)
+never inlined, which incurs a significant penalty for fast operations like `getindex`.
+Second, conversion between `Array{T}` and
+`Array{Union{Missing,T}}` [currently](https://github.com/JuliaLang/julia/issues/26681)
+involves a copy. In theory it should be possible
+to use the same memory for bits types, since only the type tag array differs.
+Third, the Julia compiler is [currently unable](https://github.com/JuliaLang/julia/issues/23336)
+to generate code using [SIMD](https://en.wikipedia.org/wiki/SIMD) instructions when `missing`
+values are involved. This is not insurmountable since masked SIMD instructions allow applying
+an operation only to some values (the non-missing ones). While the absence of SIMD reduces
+noticeably the performance of many operations, it appears that Julia already achieves
+the same speed as vectorized operations in R (which are implemented in C). So there is
```

Comment:
This is not completely true currently due to the absence of inlining, which is why I'd rather wait a bit more for performance improvements before publishing the post.